docs/getting-started/example-bpf-k8s.md

```diff
@@ -233,32 +258,36 @@ The userspace programs have been pre-built and can be found here:
 * [quay.io/bpfman-userspace/go-uprobe-counter:latest](https://quay.io/organization/bpfman-userspace)
 * [quay.io/bpfman-userspace/go-xdp-counter:latest](https://quay.io/organization/bpfman-userspace)
 
-The example yaml files below are loading from these image.
+The example yaml files below are making use of these images:
 
 * [go-kprobe-counter/deployment.yaml](https://github.com/bpfman/bpfman/tree/main/examples/config/base/go-kprobe-counter/deployment.yaml)
 * [go-tc-counter/deployment.yaml](https://github.com/bpfman/bpfman/tree/main/examples/config/base/go-tc-counter/deployment.yaml)
 * [go-tracepoint-counter/deployment.yaml](https://github.com/bpfman/bpfman/tree/main/examples/config/base/go-tracepoint-counter/deployment.yaml)
 * [go-uprobe-counter/deployment.yaml](https://github.com/bpfman/bpfman/tree/main/examples/config/base/go-uprobe-counter/deployment.yaml)
 * [go-xdp-counter/deployment.yaml](https://github.com/bpfman/bpfman/tree/main/examples/config/base/go-xdp-counter/deployment.yaml)
 
-The userspace program in a Kubernetes Deployment doesn't interacts directly with `bpfman` like it
+The userspace program in a Kubernetes Deployment doesn't interact directly with `bpfman` like it
 did in the local host deployment.
-Instead, the userspace program running on each node, if needed, reads the `BpfProgram` object
+Instead, the userspace program running on each node, if needed, reads the `ClusterBpfApplicationState` object
 from the KubeApiServer to gather additional information about the loaded eBPF program.
-To interact with the KubeApiServer, RBAC must be setup properly to access the `BpfProgram`
+To interact with the KubeApiServer, RBAC must be setup properly to access the `ClusterBpfApplicationState`
 object.
 The `bpfman-operator` defined the yaml for several ClusterRoles that can be used to access the
 different `bpfman` related CRD objects with different access rights.
-The example userspace containers will use the `bpfprogram-viewer-role`, which allows Read-Only
-access to the `BpfProgram` object.
+The example userspace containers will use the `bpfman-bpfapplication-viewer-role` and
```

Comment:
Question:

The `bpfman-operator` defined the yaml for several ClusterRoles that can be used to access the
different `bpfman` related CRD objects with different access rights.
The example userspace containers will use the `bpfman-bpfapplication-viewer-role` and
`bpfman-clusterbpfapplication-viewer-role` clusterroles, which allow Read-Only
access to the `BpfApplication` / `ClusterBpfApplication` objects.
This ClusterRole is created automatically by the `bpfman-operator`.

---> Shouldn't the bpfman-bpfapplication-viewer-role and bpfman-clusterbpfapplication-viewer-role roles also include the *State objects?
Wouldn't a userspace application get info about the eBPF application from the *State, rather than from the (Cluster)BpfApplication?